app/models/user.rb

```diff
@@ -56,21 +56,41 @@
 class User < ApplicationRecord
   devise :database_authenticatable, :lockable, :validatable, :timeoutable, :trackable, :invitable, :recoverable
 
-  belongs_to :vita_partner, optional: true
   has_many :assigned_tax_returns, class_name: "TaxReturn", foreign_key: :assigned_user_id
+
+  #
+  belongs_to :vita_partner, optional: true
   has_and_belongs_to_many :supported_organizations,
            join_table: "users_vita_partners",
            class_name: "VitaPartner"
+  #
+
+  has_many :memberships
+  accepts_nested_attributes_for :memberships
 
   attr_encrypted :access_token, key: ->(_) { EnvironmentCredentials.dig(:db_encryption_key) }
 
   validates_presence_of :name
   validates_inclusion_of :timezone, in: ActiveSupport::TimeZone.country_zones("us").map { |tz| tz.tzinfo.name }
 
+  # returns true if the user has role "lead" for organization or its parents.
+  # @param keyword [Boolean] exclude_ancestors
+  def can_lead?(vita_partner, exclude_ancestors: false)
+    memberships.where({
+        role: "lead",
+        vita_partner_id: [
+          vita_partner.id,
+          *(vita_partner.ancestors unless exclude_ancestors)
+        ].compact
+    }).exists?
+  end
```

Comment:
Why couldn't this be
```ruby
def can_lead(vita_partner)
  memberships.where(vita_partner: vita_partner, role: "lead").exists?
end
```

I couldn't find a unit test for this method. But I also couldn't find a situation in the code where we would call
```
user.can_lead?(vita_partner)
```
and expect it to return true if the user were a member on vita_partner but a lead on the vita_partner's parent organization.

Reply:
We're planning on removing that column, but I generally remove references in one PR, and then drop columns in followup PRs/tasks once we've ensured on production that the data migration worked and that everything is working okay, rather than dealing with reverting to backups etc.